src_c/polygon.c

```diff
@@ -315,6 +319,66 @@ pg_polygon_dealloc(pgPolygonObject *self)
     Py_TYPE(self)->tp_free((PyObject *)self);
 }
 
+static PyObject *
+pg_polygon_normal_polygon(PyObject *_null, PyObject *const *args,
+                          Py_ssize_t nargs)
+{
+    int sides;
+    double radius;
+    double angle = 0;
+    double Cx, Cy;
+
+    if (nargs < 3 || nargs > 4) {
+        return RAISE(PyExc_TypeError,
+                     "invalid number of arguments, expected 3 or 4 arguments");
+    }
+    if (!PyLong_Check(args[0])) {
+        goto error;
+    }
+    sides = PyLong_AsLong(args[0]);
+    if (PyErr_Occurred()) {
+        return NULL;
+    }
+    if (!pg_TwoDoublesFromObj(args[1], &Cx, &Cy) ||
+        !pg_DoubleFromObj(args[2], &radius)) {
+        goto error;
+    }
+    if (nargs == 4) {
+        if (!pg_DoubleFromObj(args[3], &angle)) {
+            goto error;
+        }
+    }
+
+    angle *= PI / 180.0;
```

Comment:
the angle calculation is only useful when the angle is passed as a parameter, you should move that into the if statement:
```suggestion
    if (nargs == 4) {
        if (!pg_DoubleFromObj(args[3], &angle)) {
            goto error;
        }
        angle *= PI / 180.0;
    }

    
```